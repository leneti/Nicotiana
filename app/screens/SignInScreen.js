import React, { useState } from "react";
import {
  Image,
  StyleSheet,
  View,
  Text,
  TextInput,
  Platform,
  Alert,
} from "react-native";
import colors from "../config/colours";
import { firebase } from "../config/firebase";
import { signInWithGoogleAsync as GoogleSignIn } from "../components/GoogleSignIn";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";
import AwesomeButton from "@umangmaurya/react-native-really-awesome-button";
import { Indicator } from "../components/LoadingIndicator";
import { Ionicons } from "@expo/vector-icons";

export default function SignInScreen() {
  const [loading, setLoading] = useState(false);
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const login = () => {
    console.log("Trying to log in");
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .catch((error) => {
        Alert.alert("Unsuccessful Login Attempt", error.message);
      })
      .finally(() => {
        setLoading(false);
      });
  };

  return (
    <View style={styles.background}>
      {loading && (
        <View
          style={{ flex: 1, justifyContent: "center", alignItems: "center" }}
        >
          <View
            style={{
              width: wp(50),
              height: hp(25),
              justifyContent: "center",
              alignItems: "center",
            }}
          >
            <Indicator />
          </View>
        </View>
      )}

      {!loading && (
        <View
          style={{
            flexDirection: "column",
            justifyContent: "center",
            alignItems: "center",
            height: hp(60),
            width: wp(75),
          }}
        >
          <Text
            style={{
              fontSize: 45,
              color: colors.pink,
              fontFamily: "Yantramanav_300Light",
              marginBottom: hp(7),
            }}
          >
            NICOTIANA
          </Text>
          <View style={styles.container}>
            <TextInput
              style={styles.text}
              underlineColorAndroid="transparent"
              placeholder="Email"
              placeholderTextColor={colors.grey}
              autoCapitalize="words"
              autoCompleteType="email"
              keyboardType="email-address"
              onChangeText={setEmail}
            />
          </View>
          <View style={styles.container}>
            <TextInput
              style={styles.text}
              underlineColorAndroid="transparent"
              placeholder="Password"
              placeholderTextColor={colors.grey}
              autoCapitalize="words"
              returnKeyType="done"
              secureTextEntry={true}
              onChangeText={setPassword}
            />
          </View>

          <AwesomeButton
            style={{ marginTop: hp(5) }}
            progress
            onPress={(next) => {
              setLoading(true);
              login();
              next();
            }}
            width={wp(70)}
            borderRadius={20}
            height={45}
            textColor={colors.white}
            textSize={16}
            backgroundColor={colors.pink}
            backgroundDarker={colors.pink_darker}
            raiseLevel={3}
          >
            Log in
          </AwesomeButton>
          <View
            style={{
              flexDirection: "row",
              justifyContent: "space-evenly",
              alignItems: "center",
              width: wp(70),
              marginVertical: hp(2),
            }}
          >
            <View
              style={{ width: wp(30), height: 1, backgroundColor: colors.grey }}
            />
            <Text style={{ color: colors.grey }}>or</Text>
            <View
              style={{ width: wp(30), height: 1, backgroundColor: colors.grey }}
            />
          </View>
          <AwesomeButton
            progress
            onPress={async (next) => {
              setLoading(true);
              const result = await GoogleSignIn();
              if (result.cancelled) setLoading(false);
              next();
            }}
            width={wp(70)}
            borderRadius={20}
            height={45}
            backgroundColor={colors.white}
            backgroundDarker={colors.grey}
            raiseLevel={3}
          >
            <Ionicons name="logo-google" size={24} color={colors.background} />
            <Text
              style={{
                color: colors.background,
                fontWeight: "bold",
                fontSize: 16,
                marginStart: wp(2),
              }}
            >
              Continue with Google
            </Text>
          </AwesomeButton>
        </View>
      )}

      <View
        pointerEvents="none"
        style={{
          alignSelf: "flex-end",
          zIndex: -1,
          bottom: 0,
          position: "absolute",
        }}
      >
        <Image
          resizeMode="contain"
          blurRadius={Platform.OS == "ios" ? 5.0 : 3.0}
          style={styles.hookah}
          source={require("../assets/hookah.png")}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  background: {
    flex: 1,
    backgroundColor: colors.background,
    justifyContent: "center",
    alignItems: "center",
  },
  container: {
    width: wp(70),
    height: 36,
    marginTop: 20,
  },
  hookah: {
    width: 200,
    height: 366,
    alignSelf: "flex-end",
  },
  signintext: {
    color: colors.white,
    alignSelf: "center",
    paddingTop: Platform.OS === "android" ? 6 : 13,
    fontSize: 15,
  },
  text: {
    color: colors.pink,
    paddingStart: 20,
    fontSize: 16,
    height: 45,
    borderWidth: 1,
    borderRadius: 12,
    borderColor: colors.background_transparent,
    backgroundColor: colors.background_transparent,
  },
});
