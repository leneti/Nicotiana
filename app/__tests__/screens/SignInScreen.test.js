import React from "react";
import renderer from "react-test-renderer";

import SignInScreen from "../../screens/SignInScreen";

describe("<SignInScreen />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SignInScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
