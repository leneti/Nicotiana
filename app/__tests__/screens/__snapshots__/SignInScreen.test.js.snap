// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#36383F",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 800.5,
        "justifyContent": "center",
        "width": 562.5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ff9baf",
          "fontFamily": "Yantramanav_300Light",
          "fontSize": 45,
          "marginBottom": 93.5,
        }
      }
    >
      NICOTIANA
    </Text>
    <View
      style={
        Object {
          "height": 36,
          "marginTop": 20,
          "width": 525,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="words"
        autoCompleteType="email"
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#acacad"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#43454B",
            "borderColor": "#43454B",
            "borderRadius": 12,
            "borderWidth": 1,
            "color": "#ff9baf",
            "fontSize": 16,
            "height": 45,
            "paddingStart": 20,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "height": 36,
          "marginTop": 20,
          "width": 525,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="words"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#acacad"
        rejectResponderTermination={true}
        returnKeyType="done"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#43454B",
            "borderColor": "#43454B",
            "borderRadius": 12,
            "borderWidth": 1,
            "color": "#ff9baf",
            "fontSize": 16,
            "height": 45,
            "paddingStart": 20,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 45,
          "marginTop": 66.5,
          "opacity": 1,
          "width": 525,
          "zIndex": 10,
        }
      }
      testID="aws-btn-content-view"
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.15)",
            "borderBottomLeftRadius": null,
            "borderBottomRightRadius": null,
            "borderRadius": 20,
            "borderTopLeftRadius": null,
            "borderTopRightRadius": null,
            "bottom": -1.5,
            "height": 42,
            "left": "1%",
            "position": "absolute",
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
            "width": "98%",
          }
        }
        testID="aws-btn-shadow"
      />
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
            },
            Object {
              "backgroundColor": "#6e4555",
              "borderBottomLeftRadius": null,
              "borderBottomRightRadius": null,
              "borderRadius": 20,
              "borderTopLeftRadius": null,
              "borderTopRightRadius": null,
              "height": 42,
              "width": 525,
            },
          ]
        }
        testID="aws-btn-bottom"
      />
      <View
        style={
          Object {
            "borderBottomLeftRadius": null,
            "borderBottomRightRadius": null,
            "borderRadius": 20,
            "borderTopLeftRadius": null,
            "borderTopRightRadius": null,
            "height": 42,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
            "width": 525,
          }
        }
        testID="aws-btn-content"
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "overflow": "hidden",
                "zIndex": 3,
              },
              Object {
                "backgroundColor": "#ff9baf",
                "borderBottomLeftRadius": null,
                "borderBottomRightRadius": null,
                "borderColor": null,
                "borderRadius": 20,
                "borderTopLeftRadius": null,
                "borderTopRightRadius": null,
                "borderWidth": 0,
              },
            ]
          }
          testID="aws-btn-text"
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.15)",
                "height": 42,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "width": 525,
              }
            }
            testID="aws-btn-active-background"
          />
          <Text
            style={
              Object {
                "color": "#f1f1f1",
                "fontFamily": null,
                "fontSize": 16,
                "fontWeight": "bold",
                "opacity": 1,
                "paddingBottom": 0,
                "paddingHorizontal": 12,
                "paddingTop": 0,
                "textAlign": "center",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": "100%",
                "zIndex": 10,
              }
            }
            testID="aws-btn-content-text"
          >
            Log in
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "alignSelf": "flex-end",
        "bottom": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
  >
    <Image
      blurRadius={5}
      resizeMode="contain"
      source={1}
      style={
        Object {
          "alignSelf": "flex-end",
          "height": 366,
          "width": 200,
        }
      }
    />
  </View>
</View>
`;
